File: backend/app/api/v1/tools.py
"""工具注册表只读 REST（内置 / MCP / Skill 元数据统一形状）。"""

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.deps import get_db
from app.modules.tools.registry import tool_registry
from app.schemas.tools import ToolsListResponse

router = APIRouter(prefix="/tools", tags=["tools"])


@router.get("", response_model=ToolsListResponse)
async def get_tools() -> ToolsListResponse:
    """返回当前内存中的工具注册表快照（轻量；不在此接口上重建以免拖死事件循环）。"""
    return tool_registry.list_tools_public()


@router.post("/refresh", response_model=ToolsListResponse)
async def post_refresh_tools_registry(
    db: AsyncSession = Depends(get_db),
) -> ToolsListResponse:
    """重读环境变量与 settings_kv 并重建快照（改 TAVILY_API_KEY 或 MCP / Skills 后由前端主动调用）。"""
    await tool_registry.refresh(db)
    return tool_registry.list_tools_public()

File: frontend/src/api/tools.ts
/**
 * 工具注册表 API 请求函数（与 API.md §7 对齐）。
 */

import { get, post } from '@/api/client'
import type { ToolsListResponse } from '@/types/tool'

/** 获取已注册工具列表（只读快照）。 */
export function getTools(): Promise<ToolsListResponse> {
  return get<ToolsListResponse>('/api/v1/tools')
}

/** 重建工具注册表并返回最新列表（环境变量或 MCP / Skills 变更后调用）。 */
export function refreshToolsRegistry(): Promise<ToolsListResponse> {
  return post<ToolsListResponse>('/api/v1/tools/refresh')
}
